### PR TITLE
Update the PWM API module documentation

### PR DESCRIPTION
**Pull request by galpeter**

### CODE REVIEW

**Comment by galpeter** [comment]
Related issue #873

File: docs/api/IoT.js-API-PWM.md [hunk 1]
```diff
@@ -17,59 +17,34 @@ The following shows PWM module APIs available for each platform.
...
-
-### `pin` <a name="pin"></a>
-* On Linux, `pin` is a pwm number which is 0 or 1.
-* On Nuttx, you have to know pin name. The pin name is defined in target board module. For more module information, please see below list.
-  * [STM32F4-discovery](../targets/nuttx/stm32f4dis/IoT.js-API-Stm32f4dis.md#pwm-pin)
-
-
-## Class: PWM <a name="pwm"></a>
-
+### `pin`
```
Comment (zherczeg) [inline docs/api/IoT.js-API-PWM.md]: What is the purpose of this section?

File: docs/api/IoT.js-API-PWM.md [hunk 2]
```diff
@@ -17,59 +17,34 @@ The following shows PWM module APIs available for each platform.
...
 
-## Constructor <a name="pwm-constructor"></a>
 
+## Class: PWM
 
-### `new PWM()` <a name="pwm-new"></a>
+### new PWM()
 
-Returns a new PWM object which can open PWM pin.
+Returns a new PWM object which can open a PWM pin.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-PWM.md]: Is there a description what PWM exactly means?

File: docs/api/IoT.js-API-PWM.md [hunk 3]
```diff
@@ -17,59 +17,34 @@ The following shows PWM module APIs available for each platform.
...
-### `pwm.open(configuration[, callback])` <a name="pwm-open"></a>
-* `configuration <Object>`
-  * `pin <Number>`, mandatory configuration
-  * `chip <Number>`, chip number, Default: 0, only Linux configuration
-  * `period <Number>`, seconds(positive integer) 
-  * `frequency <Number>`, Hz(positive integer)
-  * `dutyCycle <Number>`, between 0 and 1
-* `callback <Function(err: Error | null)>`
+### pwm.open(configuration[, callback])
+* `configuration` {Object} Configuration object which can have the following entries
```
Comment (zherczeg) [inline docs/api/IoT.js-API-PWM.md]: following properties

File: docs/api/IoT.js-API-PWM.md [hunk 4]
```diff
@@ -17,59 +17,34 @@ The following shows PWM module APIs available for each platform.
...
+  * `pin` {number} The specific pin to use with this PWM object (mandatory configuration).
```
Comment (zherczeg) [inline docs/api/IoT.js-API-PWM.md]: Specify the pin

File: docs/api/IoT.js-API-PWM.md [hunk 5]
```diff
@@ -17,59 +17,34 @@ The following shows PWM module APIs available for each platform.
...
+  * `chip` {number} The PWM chip number (only on Linux). **Default:** `0`.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-PWM.md]: Don't mix types. This should be Specify the PWM number. Or the above: The pin index of the PWM object.

File: docs/api/IoT.js-API-PWM.md [hunk 6]
```diff
@@ -17,59 +17,34 @@ The following shows PWM module APIs available for each platform.
...
+  * `period` {number} The period of the PWM signal, in seconds(positive integer).
+  * `frequency` {number} Hz(positive integer)
```
Comment (zherczeg) [inline docs/api/IoT.js-API-PWM.md]: In Hz (positive integer).

Comment (zherczeg) [inline docs/api/IoT.js-API-PWM.md]: Btw {integer} is allowed as a type. See Buffer.

File: docs/api/IoT.js-API-PWM.md [hunk 7]
```diff
@@ -110,28 +84,39 @@ pwm0.setPeriod(1, function(err) {
   }
   console.log('done');
 });
+console.log('do');
+// prints:
```
Comment (zherczeg) [inline docs/api/IoT.js-API-PWM.md]: prints is before console.log

```
// prints: do
console.log('do');
```

File: docs/api/IoT.js-API-PWM.md [hunk 8]
```diff
@@ -110,28 +84,39 @@ pwm0.setPeriod(1, function(err) {
...
+// do
+// done
 ```
 
 
-### `pwmpin.setPeriodSync(period)` <a name="pwmpin-period-sync"></a>
-* `period <Number>`, seconds(positive integer) 
+### pwmpin.setPeriodSync(period)
+* `period` {number} The period of the PWM signal, in seconds(positive integer).
```
Comment (zherczeg) [inline docs/api/IoT.js-API-PWM.md]: space before open parenthesis

File: docs/api/IoT.js-API-PWM.md (hunk 1 above)
Comment (galpeter) [inline docs/api/IoT.js-API-PWM.md]: Nothing, just a leftover. Will remove it.

**Comment by galpeter** [comment]
@zherczeg I've updated the RP

**Comment by yichoi** [comment]
LGTM

**Pull request merged**